src/values/AbstractValue.js

```diff
@@ -177,6 +177,12 @@ export default class AbstractValue extends Value {
     // Neither this nor val is a known value, so we need to some reasoning based on the structure
     if (this.equals(val)) return true; // x => x regardless of its value
     if (!val.mightNotBeTrue()) return true; // x => true regardless of the value of x
+    // x => x !== null && x !== undefined
+    if (val instanceof AbstractValue && val.kind === "!==") {
+      let [x, y] = val.args;
+      if (this.equals(x)) return y instanceof NullValue || y instanceof UndefinedValue;
```

Comment:
Expression normalization would allow cutting potentially error prone code duplications...

Reply:
Tricky, but seems correct.

Reply:
We do have normalization, so the latter check seems like dead code. Well spotted.